staging/src/k8s.io/client-go/plugin/pkg/client/auth/gcp/gcp.go

```diff
@@ -114,6 +114,9 @@ type gcpAuthProvider struct {
 }
 
 func newGCPAuthProvider(_ string, gcpConfig map[string]string, persister restclient.AuthProviderConfigPersister) (restclient.AuthProvider, error) {
+	klog.Warningf(`WARNING: in-tree gcp auth plugin is now deprecated.
+Please use the gcloud kubectl/client-go credential plugin instead.`)
```

Comment:
We should hold off on this until we have an alternative. This is not currently actionable. Can we add this to release notes instead?

Reply:
I would prefer to have the warning where an end-user would actually see it.  Perhaps GCP can make a documentation page that we can link to?  Then you can update the documentation page with details when it becomes actionable.

Also GCP has had years of runway to migrate away from this.  This should be no surprise.  It is time.

Reply:
Moved to a `.V(1)` to limit unactionable warning.

Reply:
Opened #103499 to track moving this back to a `Warningf` by 1.24 at the latest.